2015-02-04  Guan Zhou  <guan@lnxsrv01.seas.ucla.edu>

	    * timespec.h (timespec_cmp): In the function, I found we 
	    don't have to use the subtraction to compare two numbers. 
	    Subtraction sometimes causes overflows, so I use '<' and 
	    '>' instead to make the comparison. First, we compare the
	    seconds of the two files, and if seconds are the same, 
	    we can compare their nano second.


--- old_timespec.h	2015-02-04 11:38:58.430828000 -0800
+++ timespec.h	2015-02-04 14:12:13.484954000 -0800
@@ -45,8 +45,21 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  int diff;
+  if(a.tv_sec > b.tv_sec)
+    diff=1;
+  else if (a.tv_sec < b.tv_sec)
+    diff=-1;
+  else
+  {
+    if(a.tv_nsec > b.tv_nsec)
+      diff=1;
+    else if (a.tv_nsec < b.tv_nsec)
+      diff=-1;
+    else
+      diff=0;
+  }
+  return diff;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
